Add monitoring integation

guangbochen opened the pull request:
https://github.com/harvester/harvester/issues/576
Add built-in monitoring integration 

**What does this PR do:**
- Add built-in monitoring integration by leveraging the rancher monitoring chart and [kubevirt monitoring integration with the prometheus-operator](https://kubevirt.io/user-guide/operations/component_monitoring/#integrating-with-the-prometheus-operator)
- Add default Harvester monitoring dashboards, one for the dashboard page and the other for the VM instance details page.

**Testing Plan:**
1. harvester monitoring should be enabled by default when installed from the ISO
2. check if the monitoring pod are up and running
```
$kubectl get pod -n cattle-monitoring-system
NAME                                                     READY   STATUS    RESTARTS   AGE
prometheus-rancher-monitoring-prometheus-0               3/3     Running   1          3m47s
rancher-monitoring-grafana-7f54b7d8bc-8pj4c              3/3     Running   0          106s
rancher-monitoring-kube-state-metrics-744b9448f4-2tgwj   1/1     Running   0          4m4s
rancher-monitoring-operator-754bcd8cb4-w85jz             1/1     Running   0          4m4s
rancher-monitoring-prometheus-adapter-77568b975-k49n5    1/1     Running   0          4m4s
rancher-monitoring-prometheus-node-exporter-nsmzt        1/1     Running   0          4m4s
```
3. check if two built-in harvester monitoring dashboards are loaded via:
```
$kubectl get configmap -n cattle-dashboards | grep har
harvester-vm-dashboard                                 1      9m14s
harvester-vm-detail-dashboard                          1      9m14s
```
4. check prometheusrules and servicemonitors are exist
```
$ kubectl get servicemonitor -n cattle-monitoring-system
cattle-monitoring-system   prometheus-kubevirt-rules               28m
cattle-monitoring-system   rancher-monitoring-grafana              27m
cattle-monitoring-system   rancher-monitoring-kube-state-metrics   27m
cattle-monitoring-system   rancher-monitoring-node-exporter        27m
cattle-monitoring-system   rancher-monitoring-operator             27m
cattle-monitoring-system   rancher-monitoring-prometheus           27m

$kubectl get prometheusrules -n harvester-system
NAME                        AGE
prometheus-kubevirt-rules   29m
```
5. check graphs from the dashboard page and VM details page are showing correctly.
![image](https://user-images.githubusercontent.com/4569037/134501783-bd278285-1243-48ce-a642-5944ff9bd905.png)

gitlawr commented on scripts/images/rancher-images.txt:
```diff
@@ -7,3 +7,9 @@ docker.io/rancher/rancher-webhook:v0.2.0
+docker.io/rancher/mirrored-prometheus-operator-prometheus-config-reloader:v0.48.0
+docker.io/rancher/mirrored-prometheus-operator-prometheus-operator:v0.48.0
+docker.io/rancher/mirrored-prometheus-prometheus:v2.27.1
+docker.io/rancher/mirrored-kiwigrid-k8s-sidecar:1.12.2
```
Can you check if there are any missing images? AFAIK there should be a grafana image?

guangbochen replied:
added via https://github.com/harvester/harvester-installer/compare/10e7f83a6fffc2d05be82fe5ac86e669be0fea71..d8bd2be27583944064aeb158df61a5dec502c17b, thanks

gitlawr reviewed (approved):
LGTM

bk201 commented:
(Air-gapped) I got an ImagePullBackOff error on `cattle-monitoring-system/rancher-monitoring-crd-create-bwxbd` pod.
```
 NAME↑                              PF IMAGE                 READY  STATE             INIT   RESTARTS PROBES(L:R) CPU MEM CPU/R:L  MEM/R:L  %CPU/R  %CPU/L  %MEM/R  %MEM/L PORTS  AGE      │
│ create-crds                        ●  rancher/shell:v0.1.8  false  PodInitializing   false         0 off:off       0   0     0:0      0:0     n/a     n/a     n/a     n/a        3m40s    │
│ set-preserve-unknown-fields-false  ●  rancher/shell:v0.1.8  false  ImagePullBackOff  true          0 off:off       0   0     0:0      0:0     n/a     n/a     n/a     n/a        3m40s    │
```

guangbochen commented:
Updated and added via https://github.com/harvester/harvester-installer/pull/126/commits/982181fda8e5be15983db7a90ef585cd1b685db3#diff-08a2aea3ce711fa199b168518cdea1b9b77d7bdd47ba4c1174a0edf65473ab5eR20, thanks

bk201 reviewed (approved):
LGTM!
